### ISSUE TITLE
Warhammer: Vermintide 2 (552500)

### ISSUE THREAD

**Issue by h8672** [ValveSoftware/Proton#1528]
# Compatibility Report
- Name of the game with compatibility issues: Warhammer: Vermintide 2
- Steam AppID of the game: 552500

## System Information
- GPU: GTX 1060 6GB
- Driver/LLVM version: Nvidia 396.54
- Kernel version: 4.15.0-34-generic
- Link to full system information: [Gist](https://gist.github.com/h8672/b99bb02dbff7ac920f756944433f09a4)
- Proton version: 3.7-6

## I confirm:
- [ X ] that I haven't found an existing compatibility report for this game.
- [ X ] that I have checked whether there are updates for my system available.

[steam-552500.log](https://github.com/ValveSoftware/Proton/files/2404993/steam-552500.log)

## Symptoms 
- Normally game doesn't launch after installation.
- Game has a beta, which you can join, to skip the launcher.
- Open library, launch game there for a popup where you can select skip launcher option (doesn't pop up from quick launch).
- Game may crash if you change video settings many times in row.
- If game starts windowed it may start stretched to desktop.
  - Don't move window position before you got to the character selection or it may go to unreachable place.
  - Moving resizes it in Ubuntu 18.04.
  - I used alt + space to move the window and alt stayed down ingame, which caused character walk slowly, and you just have to press alt again to release it.

## Reproduction
- Install the game in steam.
- Join temporary_launcher_workaround beta for the game. 
  - Verify local files (If you started it before, maybe optional?).
- Launch game through library and select bypass launcher, click play.

**Comment by keraldi** [comment]
Things to add: [System Information Gist](https://gist.github.com/Iorce/4e6f07136954e282f181be81e9b54628)

## Symptoms
 - EAC will not be installed/will not work even if its setup is launched manually in "SteamLibrary/steamapps/common/Warhammer Vermintide 2/installers/EasyAntiCheat/"
 - due to bypassing the launcher (and EAC not being usable), progression is not saved on the game servers and trying to retrieve or save information from/to the game servers will give an in-game prompt to quit the game.
 - tinted screen after loading in (not affecting the HUD) pitch black (does not resolve itself most of the time, workaround by changing FOV, changing FOV while everything looks as expected in turn turns everything black), green and pink tints or dense "fog" resolve itself after a couple seconds, as do graphical glitches/artifacts/stripes after loading in.

**Comment by GloriousEggroll** [comment]
just another confirmation here at what @lorce said. game doesnt save progress and you cant join online matches due to easy anti-cheat

**Comment by curieos** [comment]
Are any of these issues fixable on the Proton side? I found a [thread on the fatshark forums](https://forums.fatsharkgames.com/t/easy-anti-cheat-sabotages-multiplayer-on-linux-steam-play-proton/29016) that shed some light on the issue with EasyAntiCheat. I'm ok with setting up a VM to run this game, but if it's possible I'd like to stay away from windows.

Also, I was never able to get it to launch properly. I enabled the skip launcher option and it displayed the various splash screens, but when it's supposed to show the menu it crashes. I doesn't *really* matter if a fix isn't possible, but it was a bit frustrating for me.

**Comment by h8672** [comment]
There's a thread about it in #1468 and there should be more information about the issue.

Edit:
Hmm... maybe I should keep this open as the issue isn't solved... The game doesn't work as intended either...

**Comment by romulasry** [comment]
"Play Warhammer: Vermintide 2
Play for free! Ends in 2 days"

It is also on sale.
https://store.steampowered.com/app/552500/Warhammer_Vermintide_2/?snr=1_4_4__100

I am going to check and see if it works.

**Comment by romulasry** [comment]
Latest one crashes when selecting (Bypass Launcher --selected under Betas) otherwise it doesn't load. Proton 3.16-7.

**Comment by romulasry** [comment]
Same on Proton 3.16-8.

**Comment by romulasry** [comment]
Gist: https://gist.github.com/romulasry/9963e49d07f6c7ec9f6172c7b17bfed5
Log: [steam-552500.log](https://github.com/ValveSoftware/Proton/files/2948107/steam-552500.log)
Using: Bypass Launcher

**Comment by kisak-valve** [comment]
Hello @romulasry, your log contains `err:winediag:FILE_CreateFile Too many open files, ulimit -n probably needs to be increased`, please run [`ulimit -Hn`](https://github.com/ValveSoftware/Proton/wiki/Requirements#fd-limit-requirements-systemd--240) and verify it is a high value and not 4096.

**Comment by romulasry** [comment]
Updated to systemd and system-32bit fixed it. Opensuse tumbleweed.

**Comment by curieos** [comment]
Looks like [Valve is working to get EasyAntiCheat to work on linux!](https://www.gamingonlinux.com/articles/apparently-valve-are-working-with-easy-anti-cheat-to-get-support-in-steam-play.13573) I'm pretty excited for this, V2 is one of my favorite games.

**Comment by Patola** [comment]
So, no solution for anticheat yet.

**Comment by flindeberg** [comment]
An update for the interested:
https://www.reddit.com/user/Guy1524/comments/hc3o1y/eac_progress_update/

**Comment by flindeberg** [comment]
> So, no solution for anticheat yet.

No longer true, some crazy frogs did it.

https://drive.google.com/file/d/1KWETkDj89ZCG8p5PHPqLAtgMd4X77gk6/view?usp=sharing

Might take a little while until it comes all the way to Proton though.

**Comment by Gibbz** [comment]
What is the last file linked?Whats the file provided? Does that allow this game to work multiplay?

**Comment by Patola** [comment]
@Gibbz After EAC updated, it no longer works.

**Comment by flindeberg** [comment]
> What is the last file linked?Whats the file provided? Does that allow this game to work multiplay?

@Gibbz, as @Patola said, it no longer works since the later EAC-patches.

It was a Wine-tree with EAC specific patches. As such only relevant for those of us with the knowledge and time at hand to compile wine.

Discussion takes place at #easy-anti-lobster-dev in the VKx-discord (https://discord.gg/mjWm8DK should work, it used to be the official invite link AFAIK). But to be honest, the discussion has been quite dead most of the fall and the winter so far.

**Comment by Gibbz** [comment]
Is it possible to play this via lan multiplay without using EAC in its current state?

**Comment by flindeberg** [comment]
> Is it possible to play this via lan multiplay without using EAC in its current state?

AFAIK you can play on the modded realm. I.e. without progress, but it should be playable.

**Comment by NTMan** [comment]
The game closes itself a minute after starting. During this time, you can have time to go into the settings and run the benchmark.

Demonstration: https://youtu.be/tN83gdJ35eE

## System Information
- GPU: AMD Radeon 6900XT
- Driver/LLVM version: [Mesa 21.3-git](https://gitlab.freedesktop.org/mesa/mesa/-/commit/ed6e586562f4443c56e385017a109e5a5ca5294e)/12.0.0
- Kernel version: 5.13-rc3
- Link to full system information report as [Gist](https://gist.github.com/NTMan/33cca4906328793206218192820779ec)
- Proton version: experimental-6.3-20210604

[steam-552500.log](https://github.com/ValveSoftware/Proton/files/6602851/steam-552500.log)

If you hurry up and press "Play", then after the splash with the shark, the message "Backend rejected the challenge response" will be displayed.

![Screenshot from 2021-06-05 22-17-41](https://user-images.githubusercontent.com/200750/120900005-ff1efd80-c64b-11eb-96ee-084b9caaa17c.png)

[steam-552500.log](https://github.com/ValveSoftware/Proton/files/6602940/steam-552500.log)

**Comment by flindeberg** [comment]
@NTMan The launcher did not work a while back, so they have apparently changed something (or proton / wine includes more dependencies than before). Does modded realm work for you? What is your performance compared to Windows?

**Comment by DeathTBO** [comment]
@flindeberg I recently delete my Proton prefix, and booted up the game. The launcher worked, but it will crash after ~10 seconds. Just long enough to hit the modded realm check box, and boot up the game.

It performs extremely well. ~80fps on maximum graphics at 1440p. Never goes lower than 65 fps, even when a huge horde is attacking. Highest peaks are 90.

Fedora 34 Kernel 5.13
Ryzen 2700 4ghz oc
Radeon 5700xt

Enabling mods can be done by opening the config file located at ``steamapps/compatdata/552500/pfx/drive_c/users/steamuser/AppData/Roaming/Fatshark/Vermintide 2/user_settings.config`` and setting each mod to ``enabled = true``

That folder also has a launcher.cfg where you can set ``SendCrashReports`` to false and ``DisableAntiCheat`` to true. 

Technically, you can only progress on the Official Realm, **BUT** there is a mod known as "Casual Mode". Which allows you to unlock everything straight up, or use a progression system similar to the Official Realm. It uses xp gained from the missions. It's pretty much perfect.

I found some mods were unstable, and crashed the game, so I cut it down to only the Vermintide Mod Framework, Penlight Lua Libraries, and Casual Mode (I assume load order is from top to bottom). I haven't used it very much, but it seems to work pretty well.

**Comment by Zorrototo** [comment]
Work is being done, we'll see when it will be enabled and working https://steamdb.info/depot/552503/history/?changeid=M:5995370140976663476

**Comment by DevanPratt** [comment]
Looks like they have been trying to reach out for enabling EAC support: https://steamcommunity.com/app/552500/discussions/0/2960544187643622408/?ctp=26#c3419936604888208895

**Comment by Zorrototo** [comment]
Seems like people are having some working EAC on the Steam forums https://steamcommunity.com/app/552500/discussions/0/2960544187643622408/?ctp=30#c3457094584911997892 and a dev even replied even if he still didn't confirm they changed something the past days. So far on my end, no luck, can't have anything to work. Will update if I find my issue and a fix.

**Comment by GloriousEggroll** [comment]
It's working here on Nobara + GE-Proton7-28. Clean install, clean prefix. Multiplayer works.
![Screenshot from 2022-08-04 23-34-19](https://user-images.githubusercontent.com/11287837/183009584-a01f1cbe-9606-41c4-9f2f-5f45571e9be5.png)

**Comment by DeathTBO** [comment]
Proton Experimental allowed me in.

I simply verified the game files -> Made sure Proton Experimental and the Proton EAC Runtime downloaded (I hit the queue download now) -> Bam the game let me in, I collected the daily reward.

**Comment by hjri** [comment]
i reinstalled proton eac runtime and now instead of "eac not running it says "insufficient resources"

some progress I guess.

**Comment by Zorrototo** [comment]
Yep that's where I am (see link above, I posted screenshot) since two days, I tried everything possible (I did not do a full reinstall of the 100GB files of the game, I tried at some point to delete everything but the Bundle folder, and verify the files in Steam to re-acquire the missing binaries and other files). I Tested Proton GE, Experimental (and its Bleeding Edge beta), I reset prefix all the time, I also tried to force reinstall EAC in the prefix, tried the various game beta options, tried with or without various launch options in game properties. But it seems my setup is doomed, when some other people can simply make sure to use Proton Experimental and have EAC Runtime installed, and they are all good instantly.

Maybe this is related to network and EAC can not communicate or download something, this is where my thoughts are now that I think I virtually tested everything on the Proton/game side. I might need to fully reinstall the game but to me that makes no sense when the game files verification should have made sure all files were good.

//EDIT: also, it seems it only works when you host the game as the Proton user, if you join other people's game it will kick you out after the usual EAC check.

**Comment by r2rX** [comment]
I managed to get it running once with Proton GE 7-28 and since testing Proton Experimental, even replicating the method with Proton GE again, I get the State: Untrusted error.

@Zorrototo you might be on to something. I never confirmed this when it was working but at this point in time, if I examine the loader.log file in the prefix' AppData\Roaming\EasyAntiCheat\226 folder, there are attempts to connect to anticheat servers (for wine64) and fails. If I run the URL in a browser, it downloads a file. I'm curious if placing this file in the appropriate folder would get it working again (although I don't know which folder this file is meant to go into or how it's processed).

Edit: Nevermind about the latter...each download is randomized, so it would make sense that each game instance opened up would download a unique file for that specific session.

Edit 2: Seems I was looking at the wrong folder. The correct folder is ~/.cache/com.epicgames.easyanticheat/226/ and the loader.log shows a successful connection to the server.

**Comment by Zorrototo** [comment]
There might be hope for me, I did not try yet but it smells like a good lead. https://github.com/ValveSoftware/Proton/issues/6051#issuecomment-1207123916 

Will report when I can test, I actually deleted the game a few hours ago :1st_place_medal: 

//EDIT: doesn't seem to help me.

**Comment by r2rX** [comment]
Sorry to read that, @Zorrototo . I'm running Fedora 36 KDE and glibc is 2.35. What's strange is that it worked once but not since (and it seems [I'm not the only one](https://steamcommunity.com/app/552500/discussions/0/2960544187643622408/?ctp=33#c3457094584919070088); comment # 493).

**Comment by DeathTBO** [comment]
> Sorry to read that, @Zorrototo . I'm running Fedora 36 KDE and glibc is 2.35. What's strange is that it worked once but not since (and it seems [I'm not the only one](https://steamcommunity.com/app/552500/discussions/0/2960544187643622408/?ctp=33#c3457094584919070088); comment # 493).

I'm also running Fedora 36 and KDE. I just played a round perfectly fine. It was a private match, but I got items afterwards and everything.

**Comment by cyrv6737** [comment]
I was able to play an entire game with glibc-eac and GE-Proton7-28.

What worked:
- Hosting my own match
- Playing with others, other players successfully joining mid-game
- Playing an entire match without getting kicked

What didn't work:
- Joining someone else's game. I could only host my own without getting kicked.
- Experienced a crash on DX12 mode. I only tested this once before going back to DX11. DX12 is desirable for VT2 since it is very CPU heavy.

Other things I noticed:
- DX11 mode requires you to reenable Fullscreen after alt-tab even with "Minimize on Alt-tab" disabled. This behavior was not observed in DX12 mode.
- DX12 (VKD3D) still runs better than DX11 (DXVK) just like on Windows (from what I noticed at Taal's Horn Keep)

System:
CPU: 5900x
GPU: Radeon RX 6900 XT
Mesa: mesa-git (`mesa-tkg-git 22.3.0_devel.157575.446d2039cb7-1`)

**Comment by Green-Sky** [comment]
after the proton experimental update yesterday/today the game wont launch for me anymore. both official and modded (-eac-untrusted). BUT changing to proton 7.0-4 (release candidate) works. including hosting games in official realm

system:
Ubuntu 20.04.4 lts
kernel: 5.15.0-43-generic
cpu: AMD Ryzen 9 PRO 3900
gpu: NVIDIA rtx 2070 (mobile?) Driver Version: 470.141.03 (i know its outdated)

**Comment by hjri** [comment]
i looked into logs of EAC and windows version asks for `https://download.eac-cdn.com/api/v1/games/226/client/wow64_win64/download/?uuid={uuid goes here but seems irrelevant}`
while running on proton it asks for 
`https://download.eac-cdn.com/api/v1/games/226/client/wine64/download/?uuid=...`

The first yields a 13.8MiB file that is placed in same directory where `loader.log` is, named `easyanticheat_wow64_x64.eac` but  wine64 file is a 0 byte file.

**Comment by kisak-valve** [comment]
Hello @Green-Sky, DXVK and VKD3D-Proton in Proton experimental now requires NVIDIA 510.47 or newer. This minimum driver requirement bump is not a bug.

**Comment by Any-Fuel-5635** [comment]
Game ran great, updated glibc to 2.36 along with numerous other system updates, game no longer gets beyond the "Preparing to launch Warhammer: Vermintide II..." screen on Steam. Both for steam-native and Steam Flatpak.

**Comment by Green-Sky** [comment]
> Hello @Green-Sky, DXVK and VKD3D-Proton in Proton experimental now requires NVIDIA 510.47 or newer. This minimum driver requirement bump is not a bug.

so i updated to 515, and had to re-restart steam. also switched to proton experimental. now it works again.

**Comment by r2rX** [comment]
I don't know if this actually helped getting the game working or not but I've repeated this multiple times when the state: untrusted error kicked in and has worked thus far. In the launcher, set it to modded realm and start the game. It gets stuck at the loading screen (after the warning) so I alt+F4, restart the launcher, set it back to the official realm, start the game and it worked. Just F.Y.I, if anyone uses the launcher bypass branch, the desktop shortcut ignores it (so the game needs to be started within the Steam client itself).

The game was hosted on my side, so I can't confirm being kicked when joining a game, and it ran pretty good in DX12. However, I noticed a few glitches which occurs with both DX11 (DXVK) and DX12 (VKD3D-Proton) running on a RX 6800M w/ Mesa 22.3.0-devel drivers.

When adjusting the camera, it can be positioned to expose gaps in the walls/floors (as well as a glitch with a waterfall and geometry of an arrow disappearing when being pulled):

![verm_1](https://user-images.githubusercontent.com/18622885/183268303-535c7d67-c4a2-4b73-9039-d479ad31bd99.png)

![verm_2](https://user-images.githubusercontent.com/18622885/183268338-6330aa18-14ff-4872-9476-4503695826b3.png)

![verm_3](https://user-images.githubusercontent.com/18622885/183268357-af8fd443-6f68-4aef-88d4-e619c87a9a86.png)

**Comment by kisak-valve** [comment]
Hello @r2rX, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by r2rX** [comment]
Hey @kisak-valve . Here's the Proton Log and system information you requested.

[steam-552500.log](https://github.com/ValveSoftware/Proton/files/9276679/steam-552500.log)
[System Information](https://gist.github.com/r2rX/faab80dd4c25d1344264c67b7424c911)

Concerning the graphics issue, I could create a ticket on the Mesa git and provide an apitrace, if it will help.

**Comment by kisak-valve** [comment]
Blind guess that the rendering issue is related to https://gitlab.freedesktop.org/mesa/mesa/-/issues/7018. It might be interesting to test the commit mentioned in that issue report. If not, then a separate issue report on gitlab.freedesktop.org for the mesa/RADV devs to ponder can be useful for them.

**Comment by hakzsam** [comment]
@r2rX Can you try again after pulling Mesa main again? It might be related to the regression kisak pointed out. Thanks!

**Comment by r2rX** [comment]
Hi @kisak-valve . Your deduction/intuition was on point so I wanted to let you and @hakzsam know that the [dccd6f49](https://gitlab.freedesktop.org/mesa/mesa/-/commit/dccd6f495a38196ca37edeb562a0e8f0ee9600cd) commit in the mesa driver did indeed fix the culling issue! Great job! (The waterfall shader/rendering is still slightly off).

**Comment by hjri** [comment]
restarted the game and now it works?

although running dx12 version + going to quests tab in okri's book hanged my X11 or something, had to restart sddm.

**Comment by hjri** [comment]
ok that dx12 hang was one of a kind freak accident, although game didn't want to start at first, complained again about "insufficient resources" even though only 32% of memory (total 32GiB available) was used.

**Comment by pprotonrep** [comment]
I experience random crashes when running the game (no eror messageis provided, it just closes back to desktop). They may happen within the first five minutes of gameplay or after half an hour - seems random.

Changing graphical settings or dx api (11/12) does not seem to have an effect. Only tried proton stable an experimental and the crashes happen with both.


[sysinfo.txt](https://github.com/ValveSoftware/Proton/files/9294257/sysinfo.txt)
[steam-552500.log](https://github.com/ValveSoftware/Proton/files/9294258/steam-552500.log)

**Comment by kisak-valve** [comment]
Hello @pprotonrep, 

<details><summary>These look like some lines of interest from your log:</summary>

```
[00000170:] EXCEPTION handling: System.InvalidProgramException: Unrecognizable runtime implemented method 'Microsoft.Web.WebView2.Core.Raw.ICoreWebView2Controller2:_VtblGap1_23 ()'
[00000170:] EXCEPTION handling: System.NotImplementedException: Unable to cast to Microsoft.Web.WebView2.Core.Raw.ICoreWebView2Controller2.
This may happen if you are using an interface not supported by the version of the WebView2 Runtime you are using.
For instance, if you are using an experimental interface from an older SDK that has been modified or removed in a newer runtime.
Or, if you are using a public interface from a newer SDK that wasn't implemented in an older runtime.
For more information about WebView2 versioning please visit the following: https://docs.microsoft.com/microsoft-edge/webview2/concepts/versioning
[...]

"<unnamed thread>" tid=00000170 this=03d50120 , thread handle : 0093a1b0, state : not waiting
  at Microsoft.Web.WebView2.Core.CoreWebView2Controller.get__nativeICoreWebView2Controller2 () [0x00027] in <24d555f686524c89b51135793780ebc7>:0
  at Microsoft.Web.WebView2.Core.CoreWebView2Controller.set_DefaultBackgroundColor (System.Drawing.Color) [0x00000] in <24d555f686524c89b51135793780ebc7>:0
  at Microsoft.Web.WebView2.WinForms.WebView2/<InitCoreWebView2Async>d__18.MoveNext () [0x001a0] in <c24ec392de7042178c4821a0361b519a>:0
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore/MoveNextRunner.InvokeMoveNext (object) [0x00000] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00071] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00000] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore/MoveNextRunner.Run () [0x00024] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation/<>c.<.cctor>b__7_0 (object) [0x00000] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at System.Windows.Threading.ExceptionWrapper.InternalRealCall (System.Delegate,object,int) [0x000e7] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.ExceptionWrapper.TryCatchWhen (object,System.Delegate,object,int,System.Delegate) [0x00004] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.WrappedInvoke (System.Delegate,object,int,System.Delegate) [0x00001] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.DispatcherOperation.InvokeImpl () [0x00099] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext (object) [0x00008] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00071] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00000] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object) [0x0002b] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
  at MS.Internal.CulturePreservingExecutionContext.Run (MS.Internal.CulturePreservingExecutionContext,System.Threading.ContextCallback,object) [0x00029] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.DispatcherOperation.Invoke () [0x00016] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.ProcessQueue () [0x00114] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.WndProcHook (intptr,int,intptr,intptr,bool&) [0x00061] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at MS.Win32.HwndWrapper.WndProc (intptr,int,intptr,intptr,bool&) [0x0003a] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at MS.Win32.HwndSubclass.DispatcherCallbackOperation (object) [0x00042] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.ExceptionWrapper.InternalRealCall (System.Delegate,object,int) [0x000c6] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.ExceptionWrapper.TryCatchWhen (object,System.Delegate,object,int,System.Delegate) [0x00004] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.WrappedInvoke (System.Delegate,object,int,System.Delegate) [0x00001] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.LegacyInvokeImpl (System.Windows.Threading.DispatcherPriority,System.TimeSpan,System.Delegate,object,int) [0x000fc] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.Invoke (System.Windows.Threading.DispatcherPriority,System.Delegate,object) [0x00011] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at MS.Win32.HwndSubclass.SubclassWndProc (intptr,int,intptr,intptr) [0x00154] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at (wrapper native-to-managed) MS.Win32.HwndSubclass.SubclassWndProc (intptr,int,intptr,intptr) <0x00067>
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) MS.Win32.UnsafeNativeMethods.DispatchMessage (System.Windows.Interop.MSG&) <0x00012>
  at System.Windows.Threading.Dispatcher.TranslateAndDispatchMessage (System.Windows.Interop.MSG&) [0x0001a] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.PushFrameImpl (System.Windows.Threading.DispatcherFrame) [0x0004c] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.PushFrame (System.Windows.Threading.DispatcherFrame) [0x00077] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Threading.Dispatcher.Run () [0x00006] in <3fca9819a0fd4fe2950d9c6ef1f9bd63>:0
  at System.Windows.Application.RunDispatcher (object) [0x00023] in <3f0ca28258364746b90df0026805ef93>:0
  at System.Windows.Application.RunInternal (System.Windows.Window) [0x0010d] in <3f0ca28258364746b90df0026805ef93>:0
  at System.Windows.Application.Run (System.Windows.Window) [0x00008] in <3f0ca28258364746b90df0026805ef93>:0
  at System.Windows.Application.Run () [0x00009] in <3f0ca28258364746b90df0026805ef93>:0
  at Launcher.App.Main () [0x0000b] in <e65b2ac48b8a4c599b0fa49e25dfb829>:0
  at (wrapper runtime-invoke) object.runtime_invoke_void (object,intptr,intptr,intptr) [0x00027] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0
```

</details>

This is followed by and access violation(c0000005).

<details><summary>Later in the log, there's also:</summary>

```
=================================================================
	Native Crash Reporting
=================================================================
Got a UNKNOWN while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at System.Object:__icall_wrapper_mono_gc_alloc_vector <0x00012>
	  at System.Object:AllocVector <0x0013b>
	  at System.String:SplitKeepEmptyEntries <0x00057>
	  at System.String:SplitInternal <0x002ab>
	  at System.String:Split <0x0009f>
	  at System.IO.Path:CanonicalizePath <0x000b3>
	  at System.IO.Path:InsecureGetFullPath <0x003ef>
	  at System.IO.Path:GetFullPath <0x00023>
	  at System.IO.File:GetLastWriteTime <0x0003f>
	  at System.IO.DefaultWatcher:IterateAndModifyFilesData <0x008cf>
	  at System.IO.DefaultWatcher:DoFiles <0x001cb>
	  at System.IO.DefaultWatcher:UpdateDataAndDispatch <0x0011b>
	  at System.IO.DefaultWatcher:Monitor <0x0023f>
	  at System.Threading.ThreadHelper:ThreadStart_Context <0x00084>
	  at System.Threading.ExecutionContext:RunInternal <0x00197>
	  at System.Threading.ExecutionContext:Run <0x0003b>
	  at System.Threading.ExecutionContext:Run <0x00057>
	  at System.Threading.ThreadHelper:ThreadStart <0x0003f>
	  at System.Object:runtime_invoke_void__this__ <0x0006a>
=================================================================
```

</details>

**Comment by flindeberg** [comment]
@pprotonrep Try skipping the launcher. I.e. use the "Bypass launcher" option when launching the game from steam. 

The log-entries mentioned by @kisak-valve concern components only used in the launcher as far as I know.

**Comment by pprotonrep** [comment]
@flindeberg thanks for the suggestion

I've been switching back and forth between the "bypass launcher" branch and the release branch to see if it made any difference. I get crashes with both, although they may not have the same root cause :shrug:

This is the log from a crash while running the "bypass launcher" branch on dx12 mode.
[steam-552500.log](https://github.com/ValveSoftware/Proton/files/9297082/steam-552500.log)

I could successfully run through a couple of acts on dx11 mode - but at this point I'm not sure if this was just pure coincidence. I'll do more testing on dx11 mode and report back, in case this helps anyone with similar problems in the future.

**Comment by pprotonrep** [comment]
I've run the game several times now, experiencing the same crashes, regardless of dx version.
[This would be a log of a crash while running on dx11](https://github.com/ValveSoftware/Proton/files/9307496/steam-552500-dx11.log), practically identical to the other one besides from the vkd3d entries.

If this is not a widespread problem maybe it's just something about my system - i wont spam the tread anymore. I'd happily help with further analysis if more users seem to be affected by this.

**Comment by alvord-industrial** [comment]
I continue to get the state untrusted error. I was recently able to enter the game, but after minute or so the game crashed. Upon restarting I get the state untrusted error. Deleting steamapps/compatdata/552500 and verifying integrity seemed to reset the issue, allowing me to join once again for a minute or so. But that seems to have stopped working.
 
[system info](https://gist.github.com/alvord-industrial/142d74b85fa9fad45d34af3c37cf0adb)
most recent log
[steam-552500.log](https://github.com/ValveSoftware/Proton/files/9339432/steam-552500.log)

**Comment by Zorrototo** [comment]
For all those with the Untrusted / Insufficient Resources issue, did one of you try to wipe everything including game files and start over? I usually am an advocate of never doing that instead of verifying files, but I'm leaning towards a full wipe/cleaning/reinstallation now, but for me this is a week long project due to my connection so I would prefer to gather information from others first.

**Comment by hjri** [comment]
i can say that i did NOT wipe all game data to get it working, i only ever deleted proton's wineprefix

insufficient resources seems a bit random, i feel like EAC wants to have certain amount of memory/disk space free or something, or it gets misreported somehow

**Comment by Patola** [comment]
Had the insufficient resources message when I tried to play with Proton 7.0-3.
The message disappeared when I changed to Proton Experimental.
Had the same symptoms of everyone: connecting to a random game works but disconnects after 1-2 minutes. Hosting the game works. Hope they fix it quickly.

**Comment by hjri** [comment]
fun fact - i had "insufficient resources" error once, then i started the game again **without changing proton version** and it worked fine.

**Comment by Patola** [comment]
> fun fact - i had "insufficient resources" error once, then i started the game again **without changing proton version** and it worked fine.

If that was recent, it might be due to the upgrade to 7.0-4 that's just out. It seems it's currently the same as Proton Experimental. 7.0-3 was automatically upgraded to it.

**Comment by hjri** [comment]
a week ago or so

**Comment by flindeberg** [comment]
> fun fact - i had "insufficient resources" error once, then i started the game again **without changing proton version** and it worked fine.

Proton (or rather Wine) does a lot of automagics when loading from a prefix with a different version. It is entirely possible, although impossible for me to tell without logs, that the automagics somehow broke for the first try and the second try skipped the automagics because it was not the first run with the prefix with that particular version of Wine.

Also remember that  EAC does quite a bit of automagics as well, for example updating the required dlls "sometimes" :tm: . 

If in doubt, try again or restart your computer (seriously).

**Comment by Hellhound1** [comment]
> For all those with the Untrusted / Insufficient Resources issue, did one of you try to wipe everything including game files and start over? 

I've tried this - no luck. Consistently getting the error.

Have tried everything suggested in this thread, and cannot get past the error no matter what I try. I've attached my specs and proton log in the hope someone can see something obvious!

[specs](https://github.com/ValveSoftware/Proton/files/9440932/specs.txt)
[steam-552500.log](https://github.com/ValveSoftware/Proton/files/9440934/steam-552500.log)

**Comment by Zorrototo** [comment]
Thanks for reporting. I had little hope for that but now I'm wondering what we could test to find what's wrong. Hope devs work on it more anyway as it still doesn't work properly anyway for those who can play.

**Comment by r2rX** [comment]
I don't know if this will help everyone trying to get around this resource issue but the following are the steps I have repeated reliably to get it working:

1. Ensure the Proton EasyAntiCheat Runtime is already installed.
2. Delete the Verminitide prefix (folder 552500) in .local/share/Steam/steamapps/compatdata/
3. Start the game. At this point, Steam should be installing the Windows libraries required for the game (i.e EasyAntiCheat, VC Redist, DirectX etc). Once completed, the launcher should pop up so select official realm and it goes in-game. It will eventually show the resource error...this is to be expected. Close the game (alt-F4).
4. Now run the file verification process and, assuming all goes well, it will report that one file for the game had an issue and it should download this file (around 65kb) as well as refreshing the Proton EasyAntiCheat Runtime, Steam Linux Runtime - Solder and maybe even the Proton version you're using.
5. Start the game after completion, launch into the official realm and it should work.

Whenever there is a Proton update, the prefix will either crap out and not update properly (hangs) or I need to redo steps 3 to 5. Assuming it's the former, I start from step 2.

**Comment by Hellhound1** [comment]
> 5. Start the game after completion, launch into the official realm and it should work.

Thank you for the suggestion, I followed your steps and it did re-download a file, however on trying the official realm I had the same error and still unable to get past this. Other EAC games, i.e. Apex work fine.

**Comment by r2rX** [comment]
Hmm. The last thing I can suggest, @Hellhound1 , is to enable the 'Bypass Launcher' branch under the 'Beta' properties, launch the game within Steam and run Bypass Launcher.

**Comment by kisak-valve** [comment]
Warhammer Vermintide 2 EAC Insufficient Resources Error

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6129.
@Montagge posted on 2022-08-30T01:12:23:

# Compatibility Report
- Warhammer Vermintide 2
- Steam AppID of the game: 552500

## System Information
- GPU: GTX1050ti
- Driver/LLVM version: nVidia 515.65.01
- Kernel version: 5.15.0-46-generic
- Link to full system information report as [Gist](https://gist.github.com/Montagge/5d9850a381f52390247bfdf72a662cb4):
- Proton version: Experimental
- Using Steam Native
- Vermintide 2 is not installed on the same drive as Steam (it won't fit as I use a smaller SSD for the OS and software)

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

Proton Log
[steam-552500.log](https://github.com/ValveSoftware/Proton/files/9448703/steam-552500.log)

EAC Game Launcher Log
[gamelauncher.log](https://github.com/ValveSoftware/Proton/files/9448711/gamelauncher.log)

Vermintide 2 Console Log
[console-2022-08-30-00.52.13-c555b455-3458-4466-ad49-dd8afea87841.log](https://github.com/ValveSoftware/Proton/files/9448718/console-2022-08-30-00.52.13-c555b455-3458-4466-ad49-dd8afea87841.log)


## Symptoms 
Game will launch and get to the warning.  Once I click continue a notification at the top right of the screen will appear informing me that EAC has insufficient resources, cannot continue and I am Untrusted.  The game will hang here and needs to be forced closed.

I have on two occasions gotten it to launch all the way into the game.  Once was after a Proton Experimental update pushed out to the Steam client. The other time seemed almost random as I have not been able to replicate it.

I have tried in various orders and combinations tried
1. Deleting the compatdata folder
2. Reinstalling Proton EAC Runtime
3. Reinstall Steam Linux Runtime - Soldier
4. Verifying the game files
5. Installing Proton EAC Runtime on same disk as game
6. Installing Proton EAC Runtime on same disk as Steam
7. Running the EAC executable in the compatdata to repair the EAC install
8. Tried the launcher bypass beta
9. Didn't use any beta

## Reproduction
For me it's pretty much guaranteed if you try to play the game

**Comment by flindeberg** [comment]
@Montagge What are the filesystems of the two drives?

**Comment by Montagge** [comment]
@flindeberg Both filesystems are ext4

**Comment by dragonfly-lover** [comment]
Proton experimental of 20/9/2022 claims the game is now playable. can someone verify this statement pls?
https://github.com/ValveSoftware/Proton/wiki/Changelog

**Comment by flindeberg** [comment]
> Proton experimental of 20/9/2022 claims the game is now playable. can someone verify this statement pls?

It worked perfectly a couple of weeks ago, and unless they have broken it, it should still work.

**Comment by Zorrototo** [comment]
It should work, but it doesn't for some people so test yourself.

**Comment by Montagge** [comment]
> Proton experimental of 20/9/2022 claims the game is now playable. can someone verify this statement pls? https://github.com/ValveSoftware/Proton/wiki/Changelog

Still not working for me on Ubuntu 20.04 LTS

**Comment by alvord-industrial** [comment]
Now I am experiencing a crash when other players join in-mission. Someone was able to join the hub and launch a mission just fine. 
[steam-552500.log](https://github.com/ValveSoftware/Proton/files/9641900/steam-552500.log)

**Comment by mlturner88** [comment]
Yeah, I have no issue running the game but EAC sure hates us Linux folks. As others have said, if you aren't hosting it will yeet you out of there within a minute. I also notice that it is very quick to crash if it thinks resources are tight in any way. I had to turn down graphics settings even though I was getting no stutters nor frame drops.

**Comment by r2rX** [comment]
> Yeah, I have no issue running the game but EAC sure hates us Linux folks. As others have said, if you aren't hosting it will yeet you out of there within a minute. I also notice that it is very quick to crash if it thinks resources are tight in any way. I had to turn down graphics settings even though I was getting no stutters nor frame drops.

I'm not certain if the issue, of being kicked after joining a game, is strictly due to EAC because I experience the same issue when joining a game in the modded realm as well.

**Comment by taisph** [comment]
Proton 7.0-4:
Launcher crashes as soon as I mouse-over the Play button in the Warhammer launcher.

Proton experimental:
Can launch the game but it crashes randomly a few minutes in with a crash dialog window popping up. First time during some cage transport intro, second during character creation.

Ubuntu 22.04.1 LTS
Nvidia GTX 1070, driver 515.65.01.

**Comment by Jaizu** [comment]
Hosting issue still persists in latest Proton RC as date of 24 hours before this comment.

**Comment by ptibedo** [comment]
For me I can play Kingdom modded from Proton 6.3, cant play from Kingdom official because EAC  dont run and cant reach the server. For Proton 7 version I cant play, he run but eac say :"Insufficient ressource, cannot continue".

**Comment by thorsten-passfeld** [comment]
In case you have a high core count CPU and you are experiencing (frequent) crashes, go into the launcher settings and limit worker threads to 6 at most! This is recommended by the developers and at the very least made crashes way less likely for me. Haven't crashed in 1h+ of gameplay vs. every 3 minutes before

Edit: I played for a bit more today and I can say that it's exactly what I expected. It just crashes less frequently than before. It crashed like a few minutes after completing one mission just now. Looks to be some kind of race condition between these "worker threads". Could very well be a bug related to Proton/Wine because otherwise Windows players would complain all day and call it unplayable.

**Comment by wunderbar78** [comment]
the launcher uses msedge webview and the updater of the webview runtime, that starts with the launcher in the background, crashes or doesn't finish and prevents the start of the game/launcher. (1)
if I kill the process, the launcher opens just fine, but it spawns another update process in the background. (2)
the game will start and no issues then, but if I close the game, there is that update.exe in the background which prevents the shutdown. (3)
After killing the process the game shuts down as it should.

edit: tested with proton-next, proton-experimental and GE-Proton7-41

(1)
![image](https://user-images.githubusercontent.com/60543364/203847194-163a8b1c-d02e-4184-80c9-a6d7c2d58567.png)

(2)
![image](https://user-images.githubusercontent.com/60543364/203847320-e9ebdd0d-5376-4ef7-b609-3b6d95eb4dd7.png)

(3)
![image](https://user-images.githubusercontent.com/60543364/203847527-a7101f4d-d635-43f8-bdee-f9fa95a0186a.png)

**Comment by NTMan** [comment]
> the launcher uses msedge webview and the updater of the webview runtime, that starts with the launcher in the background, crashes or doesn't finish and prevents the start of the game/launcher.

It's look like the bug that I described here: https://bugs.winehq.org/show_bug.cgi?id=53925

**Comment by LuxLucian** [comment]
Has anyone looked into running sanctioned mods on the official realm? The game boots normally and does not have any mods active when booting the official realm with mods enabled.

**Comment by rejedai** [comment]
> Has anyone looked into running sanctioned mods on the official realm? The game boots normally and does not have any mods active when booting the official realm with mods enabled.

seems what EAC problem

```
09:47:32.864 warning: [ModManager] Mod with id 1395655171ll and path Z:\home\exnt\.local\share\Steam\steamapps\workshop\content\552500\1395655171\mod.bin can't be added as mod hash catalog to EAC, skipping
09:47:32.864 warning: [ModManager] Mod with id 1389872347ll and path Z:\home\exnt\.local\share\Steam\steamapps\workshop\content\552500\1389872347\mod.bin can't be added as mod hash catalog to EAC, skipping
09:47:32.865 warning: [ModManager] Mod with id 1397265260ll and path Z:\home\exnt\.local\share\Steam\steamapps\workshop\content\552500\1397265260\mod.bin can't be added as mod hash catalog to EAC, skipping
09:47:32.865 warning: [ModManager] Mod with id 1369573612ll and path Z:\home\exnt\.local\share\Steam\steamapps\workshop\content\552500\1369573612\mod.bin can't be added as mod hash catalog to EAC, skipping
09:47:32.865 warning: [ModManager] Mod with id 1445717962ll and path Z:\home\exnt\.local\share\Steam\steamapps\workshop\content\552500\1445717962\mod.bin can't be added as mod hash catalog to EAC, skipping
09:47:32.871 [Lua] <mod_handles>
09:47:32.871 [Lua] </mod_handles>
09:47:32.871 [Lua] user_setting.mods =>
09:47:32.871 [Lua] [ModManager][warning] Mod "Vermintide Mod Framework" with id 1369573612 was not found in the workshop folder.
09:47:32.871 [Lua] [ModManager][warning] Did you try loading an unsanctioned mod in Official?
09:47:32.871 [Lua] [ModManager][warning] Mod "Simple UI (Abandoned)" with id 1389872347 was not found in the workshop folder.
09:47:32.871 [Lua] [ModManager][warning] Did you try loading an unsanctioned mod in Official?
09:47:32.871 [Lua] [ModManager][warning] Mod "Numeric UI" with id 1397265260 was not found in the workshop folder.
09:47:32.871 [Lua] [ModManager][warning] Did you try loading an unsanctioned mod in Official?
09:47:32.871 [Lua] [ModManager][warning] Mod "Bot Improvements - Combat" with id 1395655171 was not found in the workshop folder.
09:47:32.871 [Lua] [ModManager][warning] Did you try loading an unsanctioned mod in Official?
09:47:32.871 [Lua] [ModManager][warning] Mod "Loadout Manager" with id 1445717962 was not found in the workshop folder.
09:47:32.871 [Lua] [ModManager][warning] Did you try loading an unsanctioned mod in Official?
```

**Comment by darman96** [comment]
I just installed the game and get the "backend rejected the challenge response" error message right after starting.
It seems that EAC isn't running, do I have to start/install it manually?

**Comment by cecton** [comment]
@darman96 there is a problem if EAC runtime gets installed automatically by the game. What I do to work around this issue consistently is:

1. Exit Steam (make sure no process is living)

2. Uninstall manually the EAC runtime using the command:

	```
	# DO NOT COPY-PASTE RANDOM COMMAND OF RANDOM PEOPLE ON THE INTERNET
    # this depends on your installation, you need to correct the paths
	# rm -fR ~/.local/share/Steam/steamapps/appmanifest_1826330.acf ~/.local/share/Steam/steamapps/common/Proton\ EasyAntiCheat\ Runtime/
	```

3. Restart Steam
4. Show tools

    ![image](https://user-images.githubusercontent.com/3018448/209958869-c74e884e-bb49-4c10-8d65-0a06dda2e946.png)

5. Install `Proton EasyAntiCheat Runtime`.
6. Verify game files of Vermintide 2
7. (Run game with bypass launcher)

(I think you need to do that every time there is an update of Proton... so once in a while)

**Comment by r2rX** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/1528#issuecomment-1367318785

In my experience, uninstalling Proton EasyAntiCheat isn't necessary. As long as it's installed, that's fine at least. I'm leaning to the issue being related to the Windows binary of EAC. When a prefix is updated, the Windows EAC binary needs to be reinstalled (that's why it's assumed the file integrity check fixes it...it's because the Windows EAC binary being re-installed in the process).

**Comment by h8672** [comment]
The game works fine when hosting for me currently, when joining a friend it tries to host migrate after... even if it's linux host aswell.

Found a line in a log where host breaks object, that he owns, and it removes the object (host) and server disconnects... from my client.
```
13:12:32.972 [Lua] [GameNetworkManager] game_object_destroyed_player, game_object_id=0 owner_peer_id=110000101a4646d peer_id=110000101a4646d
13:12:32.972 [Lua] PlayerManager:remove_player peer_id=110000101a4646d 1
13:12:32.974 [Lua] [GameNetworkManager] removing peer_id=110000101a4646d local_player_id=1
13:12:32.974 [Lua] [GameNetworkManager] game object was destroyed id=0 with type=player, object_destroy_func=game_object_destroyed_player, owned by peer=110000101a4646d
13:12:32.974 [Lua] [GameNetworkManager] Game session disconnected, leaving game...
```
It happens 4 times in same log, 4 times I probably connected to that game friend hosted.

[console-2022-12-31-11.58.36-47cc66d4-37e8-4961-b299-0282e7656b15.zip](https://github.com/ValveSoftware/Proton/files/10327617/console-2022-12-31-11.58.36-47cc66d4-37e8-4961-b299-0282e7656b15.zip)

Edit: I suppose it should work like that once it goes wrong...?
Edit2: Ah forgot to mention... when getting kicked out, client tries to reconnect, but it tries to connect to one of the other users and not the host of that session. The host id seems to go nil the moment it's deleted or smt...

**Comment by FSKiller** [comment]
If someone is interested the DC bug still happens.

Maybe if we look at the Source code? The game has a avid mod scene for the modded realm and the Source code of the game is public so modders have a easier time making mods. Maybe someone will have a clue?

https://github.com/Aussiemon/Vermintide-2-Source-Code

**Comment by LepaBrenaa** [comment]
Anyone found out why the game works only if you host the match? If you join someone's game, EAC kicks you out.

**Comment by Zorrototo** [comment]
This is probably because of the implementation they made (on the dev side) of EAC / Proton. They probably didn't do it right. 

//EDIT: for me it doesn't even work at all, I can't even reach my lobby.

**Comment by LepaBrenaa** [comment]
> This is probably because of the implementation they made (on the dev side) of EAC / Proton. They probably didn't do it right.

But they are the same devs from Darktide. In Darktide it works normally.

**Comment by esotericist** [comment]
> > This is probably because of the implementation they made (on the dev side) of EAC / Proton. They probably didn't do it right.
> 
> But they are the same devs from Darktide. In Darktide it works normally.

darktide uses a different networking model, and there's a number of other engine changes; wouldn't be surprised if the eac connected parts of things were different between the games.

**Comment by Green-Sky** [comment]
Someone dig into the lua and figure it out.

**Comment by LuxLucian** [comment]
Recently swapped from EndeavorOS to Manjaro and trying to run the launcher returns a large error window. Bypassing the launcher results in a EAC issue.
![image](https://github.com/ValveSoftware/Proton/assets/63475999/1269dfc9-66e4-4049-9d3a-49bfb3203074)

**Comment by Green-Sky** [comment]
@LuxLucian did you make sure to install the EAC runtime on your new install?

**Comment by LuxLucian** [comment]
> @LuxLucian did you make sure to install the EAC runtime on your new install?
I am pretty sure it's installed, I also deleted the folder and verified files.
![image](https://github.com/ValveSoftware/Proton/assets/63475999/4c11552e-af75-4dd0-9f91-1ebe0258ab1e)

EDIT:
Vermintide 2 claims that EAC is not running, I'll get a proton log and upload it in another edit
![image](https://github.com/ValveSoftware/Proton/assets/63475999/6cd09e29-cc57-4a95-8202-20197199bc8f)

EDIT2: Proton logs
Log from bypass launcher (DX11)
https://pastebin.com/1x4VDRvL

Log from trying to start the launcher
https://pastebin.com/faqsNvht

**Comment by cecton** [comment]
Check this, it might help https://github.com/ValveSoftware/Proton/issues/1528#issuecomment-1367318785

**Comment by LuxLucian** [comment]
> Check this, it might help [#1528 (comment)](https://github.com/ValveSoftware/Proton/issues/1528#issuecomment-1367318785)

It didn't work (I think) but another post seems to have fixed it.

I ran the launcher and tried starting the modded realm. The game would hang on the load screen and I closed it. After running the official realm worked without the EAC issue.

EDIT: It stopped working again and this fix no longer works for me

**Comment by LuxLucian** [comment]
GPU: AMD 5700 XT
Drivers: Mesa 24.0.2-2\
Kernel:Zen 6.7.8
Proton: GE, though 9.0 (Beta) and 8.0-5 do not work either

New update seems to have broken the game again for me.
I get a popup saying "EAC state: untrusted" and it either hangs at the new login/load screen or gives a backend error and prompts me to quit the game.
I'm using these launch args

%command% : ./fix.sh

Which runs this script

 #!/bin/bash
awk -i inplace '{if ($1 ~ /StateFlags/) sub(/4/, "100"); print;}' ../../../appmanifest_552500.acf

[steam-552500.log](https://github.com/ValveSoftware/Proton/files/14548024/steam-552500.log)

It seems deleting the EAC folder in the game directory, verifying files, and resetting the prefix sometimes works.

**Comment by AjBreidenbach** [comment]
Had it partly working in the tutorial with experimental and temporary_launcher_workaround.  I didn't get to test multiplayer.  After crashing in the tutorial and relaunching I am experiencing the hanging login.

**Comment by matoro** [comment]
Was able to play one long session (hosting) on first install with no problems besides killing Edge installer during startup.

On second session received the following error:
![maim_1714530232](https://github.com/ValveSoftware/Proton/assets/12038583/99d87e9c-df5a-429e-8ca0-10bccf2d1fa0)
after which loading would hang on "Stocking Emporium" and go no further.

I tried all of the following:
* Switch to temporary_launcher_workaround beta.  I tried both "Skip Launcher" and "Skip Launcher (DX12)".
* Apply `fix.sh` script from above
* Uninstall and reinstall Proton EAC runtime
* Verify files before and after each attempt
* Delete EasyAnticheat files from game folder and recreate prefix
* Completely uninstall and redownload game

Additionally I tried every combination of these steps with each other, as well as every combination while verifying files before/after and deleting/recreating prefix each time.  Nothing worked; I am completely unable to play the game anymore.

**Comment by LuxLucian** [comment]
> Recently swapped from EndeavorOS to Manjaro and trying to run the launcher returns a large error window. Bypassing the launcher results in a EAC issue. ![image](https://private-user-images.githubusercontent.com/63475999/292803529-1269dfc9-66e4-4049-9d3a-49bfb3203074.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxMjY1NzIsIm5iZiI6MTcxNTEyNjI3MiwicGF0aCI6Ii82MzQ3NTk5OS8yOTI4MDM1MjktMTI2OWRmYzktNjZlNC00MDQ5LTlkM2EtNDliZmIzMjAzMDc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA3VDIzNTc1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0YWIzZmEzZTZhYjI0YmU0NDkwZmFlOWJkMzZkMjYxZTY0YTgzYWE1NDhmMWFlMTBmYjIyNmRlNjg0ZDU0Y2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PA2-N428flkbuWz2BIN07lTpzG3eqvZeidH-XoqVXDA)

Trying the boot the game with the launcher gives me this issue. I cannot get EAC to work and I have no way of starting the modded realm.

![image](https://github.com/ValveSoftware/Proton/assets/63475999/2bd62c0e-0769-4124-b6e1-dd43b29666ae)

**Comment by Mondrethos** [comment]
A friend of mine I regularly play with moved from Windows to Linux, (I've been on Linux all this time) and to play together I always hosted. Now that we are both on Linux, we can't play together.

Any resolution or insight into this weird issue?

**Comment by Zorrototo** [comment]
No, the person on Linux MUST host.
Nothing has changed since the initial "compatibility" with Proton, it was broken and will stay broken.

**Comment by r2rX** [comment]
Unfortunately, it's still not working properly and the only ones who can fix it are the game devs (Fatshark) themselves....and it doesn't look like they will.

**Comment by Mondrethos** [comment]
That's unfortunate.

**Comment by majuss** [comment]
So what if I just want to play on the modded realm? The Launcher is not functional for me on GE Proton 9.37. I can't click anything. I want to play with someone on macOS where EAC is not available at all (otherwise the game runs fine on macOS on modded realm).

**Comment by s3r8s** [comment]
Fyi, Fatshark updated the game to use epic online services for the easy anti cheat solution, which finally fixes the linux user must host problem.

The launcher is still barely functional, and the launcher bypass beta should still be used.

I played a game with one linux user as host and both a windows and linux user as clients, worked like a charm.

**Comment by LuxLucian** [comment]
Even with the new EAC update I am still getting EAC untrusted errors when booting the game.

![Image](https://github.com/user-attachments/assets/93090e98-9f61-4e63-807b-8c3ec145dd6f)

**Comment by s3r8s** [comment]
If the untrusted state happens on an existing installation, I would suggest trying to remove the proton prefix and letting a new one be created (by removing/renaming the prefix or just switching to and from a different proton version).

One of my friends had the same issue on an existing install after the update, and making a new prefix seems to have solved it (he did also verify game files, but this did not seem to fix anything)

**Comment by LuxLucian** [comment]
I did try both things, file verifying and prefix deletion. I'll try different proton versions next.

EDIT: No dice, wiping prefix and swapping proton versions does nothing.

EDIT 2: Using the skip launcher beta seems to have worked, and I can also play on the modded realm is EAC doesn't work. I did close both vesktop and my browser just in case too. Never seen a game work so inconsistently on linux.

**Comment by kisak-valve** [comment]
Warhammer: Vermintide 2 fails to launch on Linux due to Easy Anti-Cheat (Proton Experimental, Nobara)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9094.
@01Zanon01 posted on 2025-10-11T07:38:20:

Description :
Hello Proton team,

I am experiencing an issue running Warhammer: Vermintide 2 on Linux using Steam Play with Proton Experimental. The game fails to launch due to Easy Anti-Cheat, showing an error message about "corrupted files" even though the installation is correct. It seems the game is not currently whitelisted for Linux/Proton.

Steps to reproduce:

Install Vermintide 2 via Steam on Linux (Nobara 42, KDE Plasma).

Enable Steam Play with Proton Experimental.

Launch the game.

The game fails to start, displaying a "corrupted files" message from Easy Anti-Cheat.

Expected behavior:
The game should launch successfully with Easy Anti-Cheat enabled.

System information:

Distribution: Nobara Linux 42 (64-bit)

Desktop environment: KDE Plasma 6.4.4

Kernel: 6.16.4-201.nobara.fc42.x86_64

CPU: Intel Core i5-14400F (16 threads)

GPU: NVIDIA GeForce RTX 4060 Ti 16GB

Driver: NVIDIA 580.95.05

Steam version: 1.0.0.83-4.fc42.i686

Proton version: Proton Experimental

Game ID: 552500 (Steam)

Additional information:

The game works on Windows without issues.

I understand that official support for Nobara Linux is not provided, but I would like to submit this issue so that future Proton updates may include support or a workaround for Easy Anti-Cheat.

Any guidance or testing suggestions are welcome.

Logs / Output (optional):

If possible, I can provide Steam logs (~/steam.log) showing Easy Anti-Cheat messages.

Thank you for your time and support!

**Comment by LegoChelik** [comment]
If you have a connection error to the lobby due to a supposedly disabled anti-cheat, first try the solutions listed on [ProtonDB](https://www.protondb.com/app/552500). If they don't help, enter `ping modules-cdn.eac-prod.on.epicgames.com` in the console, copy the IP address from the output, and add a line like `<IP-address> modules-cdn.eac-prod.on.epicgames.com` to /etc/hosts. This method helped me solve the problem with the lobby host/connection to other lobbies. The solution was found on the [game's forum](https://forums.fatsharkgames.com/t/cant-join-another-players-game-and-other-players-cant-join-my-host-official-realm/).

The only inconvenience is that other games with EAC, for example, The First Descendant (2074920), can only be launched with 0.0.0.0 modules-cdn.eac-prod.on.epicgames.com in /etc/hosts. I would like to see some kind of launch parameter (?) or a complete fix for the error. For now, I'm just using a simple script to switch the entry in /etc/hosts.

**Comment by cecton** [comment]
> The game fails to start, displaying a "corrupted files" message from Easy Anti-Cheat.

I tested the game again these days (on PC on SteamOS) because I haven't play for a while.

Last time (2022), I could only host otherwise I would get kicked by the EAC, I cannot use mods at all, can't even start modded realm. The launcher is not working at all.

Now (2025) the game has new launch settings in the beta version "skip launcher". I can now run the modded realm if I want to or select DX11 or DX12 easily. I don't get kicked by EAC anymore when join other's session. So there is improvement.

I tried to remove the skip launcher and enable some mods. Nowadays the launcher works better, it's glitched but quite usable. I can enable sanctioned mods and get into the settings and all. Unfortunately if I start the game with sanctioned mod I get the message "corrupted file something something the the "verify files" process" and the game quits.

**Comment by 01Zanon01** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/1528#issuecomment-3625638538

Update from 01Zanon01

I finally managed to get Warhammer: Vermintide 2 running on Nobara Linux, including online play with Easy Anti-Cheat.
Here are the steps that solved the issue for me:

1. Select a stable Proton version

In Steam → Properties → Compatibility, I forced the game to use Proton 9.x instead of Proton Experimental.

2. Opt into the “skip launcher” beta

In the Betas tab, I selected the beta branch that provides the new Bypass Launcher options.

3. Launch using the bypass launcher (no mods)

From the Steam launch menu, I chose:

Bypass Launcher
(the non-modded option, without using the official game launcher)

With these three steps, the game launched correctly and I was able to play an online session without being blocked by EAC.

Special thanks to @cecton
Your explanation about improvements in the skip launcher beta and the issues with mods/EAC helped me understand what was going wrong. Using the bypass launcher without mods solved the problem on my system.

Thanks everyone for the support!

**Comment by SerialSeller** [comment]
@01Zanon01 

I can confirm this method works on CachyOS as well.

**Comment by Taranchul** [comment]
Can Linux players finally host a game or still only join without being kicked after a few minutes?

**Comment by r2rX** [comment]
> Can Linux players finally host a game or still only join without being kicked after a few minutes?

Hosting and joining work fine, as long as you launch the game with the 'skip launcher' option.

**Comment by GloriousEggroll** [comment]
I've made a patch that allows the launcher + EAC and all game modes to work:

[0001-HACK-Force-Vermintide-2-current-working-directory-to.patch](https://github.com/user-attachments/files/25310620/0001-HACK-Force-Vermintide-2-current-working-directory-to.patch)

Problem and resolution explanation:

Vermintide 2's launcher runs from game-folder/launcher/Launcher.exe, it then launches the EAC versions of the game using '../start_protected_game.exe' <args>. 

The problem is that when it's launched this way it keeps the current working directory set as game-folder/launcher, while the start_protected_game.exe and EAC files reside in game-folder/, thus start_protected_game.exe and subsequently EAC expect to read files from game-folder/, -not- game-folder/launcher/. Since the current working directory is set as game-folder/launcher by the launcher, it breaks EAC's file reading.

This patch checks for vermintide 2 if 'start_protected_game.exe' is called then forces the current working directory to game-folder/ instead of game-folder/launcher, allowing EAC to work when run from the launcher.

**Comment by GloriousEggroll** [comment]
Unfortunately per discussion with proton devs this patch won't be accepted because it is indeed a hack and the underlying problem of 'current working directory' resolving properly either in wine or in the EAC linux bridge is what really needs to be addressed, so I wouldn't go expecting it to be upstreamed, but at least for now we have some method of working around the bug.

**Comment by GloriousEggroll** [comment]
ayyy.. looks like we got an official fix now in experimental!:

https://github.com/ValveSoftware/wine/commit/c5a27304e01e5e5f38b82974b29ca30c532ead4c

**Comment by Feynt** [comment]
> ayyy.. looks like we got an official fix now in experimental!:
> 
> [ValveSoftware/wine@c5a2730](https://github.com/ValveSoftware/wine/commit/c5a27304e01e5e5f38b82974b29ca30c532ead4c)

Installed bleeding edge experimental. Unfortunately running the launcher bypass will briefly seem to start the EasyAntiCheat_EOS.exe program, but it almost immediately closes before Vermintide 2 starts up. I believe the issue is that EAC EOS executable closing that's causing the connectivity issues on official. I'm able to "connect" to others games, get the manifest of heroes that were picked, be told to swap to something else at times, but after a lengthy connection period I'll get told I'm untrusted and get booted back to the title screen. Same with people trying to connect to me. They attempt to connect, I see the attempt, and then they get booted.

Log attached if it'll help. Shows Proton version `experimental-bleeding-edge-10.0-314083-20260217-pa57496-wc5a273-d8e6a2d-vd6ac0a`

```
Operating System: Bazzite 43
KDE Plasma Version: 6.5.5
KDE Frameworks Version: 6.22.0
Qt Version: 6.10.1
Kernel Version: 6.17.7-ba25.fc43.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5800X3D 8-Core Processor
Memory: 32 GiB of RAM (31.3 GiB usable)
Graphics Processor: AMD Radeon RX 6900 XT
Manufacturer: Gigabyte Technology Co., Ltd.
Product Name: X470 AORUS ULTRA GAMING
```

[steam-552500.log](https://github.com/user-attachments/files/25381566/steam-552500.log)

**Comment by KrauseStefan** [comment]
That build is too old, the Wine repo submodule in Proton was updated ~4H ago which includes the changes in commit https://github.com/ValveSoftware/Proton/commit/e784a4d5cbdf513870f17d283e426aace116583b.
The version should be `experimental-bleeding-edge-10.0-315014-20260218-pa57496-wc5a273-d8e6a2d-vd6ac0a` or newer.

**Comment by Feynt** [comment]
> That build is too old, the Wine repo submodule in Proton was updated ~4H ago which includes the changes in commit [e784a4d](https://github.com/ValveSoftware/Proton/commit/e784a4d5cbdf513870f17d283e426aace116583b). The version should be `experimental-bleeding-edge-10.0-315014-20260218-pa57496-wc5a273-d8e6a2d-vd6ac0a` or newer.

New version: experimental-bleeding-edge-10.0-315206-20260218-pa57496-wb49a74-dd222d2-vd6ac0a
Still unable to connect to games

[steam-552500.log](https://github.com/user-attachments/files/25403619/steam-552500.log)

**Comment by GloriousEggroll** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/1528#issuecomment-3918800750

no. dont use the launcher bypass branch.
use the normal branch
make sure you have the latest version of bleeding edge
press play, the LAUNCHER will open.
press play on the launcher.

the fix was for running the game USING THE LAUNCHER. not bypassing it.

**Comment by KrauseStefan** [comment]
I did a few games yesterday with the new build, worked like a charm 😄 🎊 

You can see the version of Steam Experimential with:
`cat ~/.local/share/Steam/steamapps/common/Proton\ -\ Experimental/version`

Posting a few dumps of the needed configs just to make it easier for everyone to test it.
<img width="600" alt="Image" src="https://github.com/user-attachments/assets/f0db3c51-ef13-40ec-a454-e5ef5e3ae97d" />

<img width="600"  alt="Image" src="https://github.com/user-attachments/assets/0bb9d6e6-5175-4600-9356-0bbc7de87706" />

<img width="600"  alt="Image" src="https://github.com/user-attachments/assets/ea87a01f-6dc1-419c-bea1-c4aee8bcce1f" />

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/93394917-cdbf-44ce-bdb9-5b07dca4e3e6" />

**Comment by Feynt** [comment]
Still no love on my end. I've updated the experimental build daily since my last post, but I'm rejected from joining games every time. The reported version is: `1771720628 experimental-bleeding-edge-10.0-316698-20260222-pa57496-w2ab7e7-d4357d6-v9ced67`

**Comment by Feynt** [comment]
Testing Vermintide 2 on an old laptop, I was able to get it to run the EAC front end. At least, I'm certain it worked, because it actually got to 100% rather than flickering on screen and disappearing. It didn't actually get into game (black screen), but the laptop is old enough that I expect it doesn't support the game.

Since both are running Bazzite, is there some way to "diff" the two systems to see what might be installed that's different? Besides me running an Radeon 6900 XT in my primary computer and the laptop having an nVidia integrated card that is. I'm curious what is making EAC say no to my main computer while the laptop is fine.
